behaviour/graql/language/undefine.feature

```diff
@@ -29,215 +29,1173 @@ Feature: Graql Undefine Query
+      match
+        $x has name;
+        $x type person;
+      get;
       """
-    Then concept identifiers are
-      |           | check | value     |
-      | EMPLOYEE  | label | employee  |
-      | PART_TIME | label | part-time |
-    Then uniquely identify answer concepts
-      | x         | role     |
-      | PART_TIME | EMPLOYEE |
+    Then answer size is: 0
 
-  @ignore
-  # re-enable when 'relates' is bound to a relation and blockable
-  # TODO
-  Scenario: undefine 'relates' from super relation that is overriden using 'as' removes override from child (?)
 
-  @ignore
-  # TODO
-  Scenario: undefine a sub-role using 'as' removes sub-role from child relations
+  Scenario: attempting to undefine an attribute ownership that was not actually owned to begin with throws an error
+    Then graql undefine throws
+      """
+      undefine employment has name;
+      """
+    Then the integrity is validated
 
-  Scenario: undefine 'plays' from super relation removes 'plays' from child relation
 
-  Scenario: undefine 'has' from super relation removes 'has' from child relation
+  Scenario: attempting to undefine an attribute ownership inherited from a parent throws an error
+    Given graql define
+      """
+      define child sub person;
+      """
+    Given the integrity is validated
+    Then graql undefine throws
+      """
+      undefine child has name;
+      """
+    Then the integrity is validated
 
-  Scenario: undefine 'key' from super relation removes 'key' from child relation
 
-  Scenario: undefine 'plays' from super attribute removes 'plays' from child attribute
+  Scenario: undefining a key ownership removes it
+    Given get answers of graql query
+      """
+      match
+        $x key email;
+        $x type person;
+      get;
+      """
+    Given concept identifiers are
+      |     | check | value  |
+      | PER | label | person |
+    Given uniquely identify answer concepts
+      | x   |
+      | PER |
+    When graql undefine
+      """
+      undefine person key email;
+      """
+    Then the integrity is validated
+    When get answers of graql query
+      """
+      match
+        $x key email;
+        $x type person;
+      get;
+      """
+    Then answer size is: 0
 
-  Scenario: undefine 'has' from super attribute removes 'has' from child attribute
 
-  Scenario: undefine 'key' from super attribute removes 'key' from child attribute
+  Scenario: attempting to undefine a key ownership that doesn't exist throws an error
+    Then graql undefine throws
+      """
+      undefine employment key email;
+      """
+    Then the integrity is validated
 
 
-  @ignore
-  # TODO fails since undefining an abstract removes the type fully
-  Scenario: undefine a type as abstract converts an abstract to concrete type and can create instances
-    Given graql undefine
+  Scenario: attempting to undefine an attribute owned with `key` by using `has` throws an error
```

Comment:
although by our new "assertion" style logic this should be now a valid operation to remove the key

Reply:
yeah

Reply:
I'll raise this in tomorrow's meeting

Reply:
We decided to ignore this test and defer the discussion.